salt/_states/metalk8s_etcd.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+import logging
+
+log = logging.getLogger(__name__)
+
+
+__virtualname__ = 'metalk8s_etcd'
+
+
+def __virtual__():
+    if 'metalk8s_kubernetes.ping' not in __salt__:
+        return False, '`metalk8s_kubernetes.ping` not available'
+    else:
+        return __virtualname__
+
+
+def member_present(name, peer_urls):
+    """Ensure that the etcd node is in cluster"""
+    ret = {'name': name,
+           'changes': {},
+           'result': False,
+           'comment': ''}
+
+    # Check that peer urls does not exist
+    if __salt__['metalk8s_etcd.urls_exist_in_cluster'](peer_urls):
+        ret['result'] = True
+        ret['comment'] = "Peer URLs: {} already exists".format(
+            ' ,'.join(peer_urls)
+        )
+        return ret
+
+    # Add node
+    member = __salt__['metalk8s_etcd.add_etcd_node'](peer_urls)
```

Comment:
Given below the `result` is always set to `True`, I guess this call can raise.

Rather than raising and relying on that, consider catching the exception and handle `result` and such accordingly.